docs/en/sql-reference/10-sql-commands/00-ddl/02-user/04-user-set-2nd-roles.md

```diff
@@ -0,0 +1,135 @@
+---
+title: SET SECONDARY ROLES
+sidebar_position: 6
+---
+
+Activates all secondary roles for the current session. This means that all secondary roles granted to the user will be active, extending the user's privileges. For more information about the active role and secondary roles, see [Active Role & Secondary Roles](/guides/security/access-control/roles#active-role--secondary-roles).
+
+Seel also: [SET ROLE](04-user-set-role.md)
+
+## Syntax
+
+```sql
+SET SECONDARY ROLES { ALL | NONE }
+```
+
+| Parameter | Default | Description                                                                                                                                                                                     |
+|-----------|---------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| ALL       | Yes     | Activates all secondary roles granted to the user for the current session, in addition to the active role. This enables the user to utilize the privileges associated with all secondary roles. |
+| NONE      | No      | Deactivates all secondary roles for the current session, meaning only the active role's privileges are active. This restricts the user's privileges to those granted by the active role alone.  |
+
+## Examples
+
+This example shows how secondary roles work and how to active/deactivate them.
+
+1. Creating roles as user root.
+
+First, let's create two roles, `admin` and `analyst`:
+
+```bash
+root@localhost:8000/default> CREATE ROLE admin;
+
+CREATE ROLE admin
+
+0 row written in 0.017 sec. Processed 0 row, 0 B (0 row/s, 0 B/s)
+
+root@localhost:8000/default> CREATE ROLE analyst;
```

Comment:
It's better to remove all `root@localhost:8000/default`, to enhance unity between on-prem and cloud.